src/web_interface/components/dependency_graph.py

```diff
@@ -13,17 +15,36 @@ def create_data_graph_nodes_and_groups(data, whitelist):
     groups = []
 
     for file in data:
-        if file['processed_analysis']['file_type']['mime'] in whitelist:
+        mime = file['processed_analysis']['file_type']['mime']
+        if mime not in whitelist or root_uid not in file['virtual_file_path']:
+            continue
+
+        if mime not in groups:
+            groups.append(file['processed_analysis']['file_type']['mime'])
+
+        virtual_paths = file['virtual_file_path'][root_uid]
+
+        for vpath in virtual_paths:
+
+            path_components = split_virtual_path(vpath)
+
+            if len(path_components) < 2:
+                continue
+
+            name_component = path_components[-1]
+            parent_component = path_components[-2]
+
+            if parent_component != parent_uid:
+                continue
+
             node = {
-                'label': file['file_name'],
-                'id': file['_id'],
+                'label': name_component,
+                'id': vpath,
+                'entity': file['_id'],
                 'group': file['processed_analysis']['file_type']['mime'],
```

Comment:
can also be replaced:
```suggestion
                'group': mime,
```